solr/core/src/java/org/apache/solr/update/processor/DistributedZkUpdateProcessor.java

```diff
@@ -883,8 +836,7 @@ private void checkReplicationTracker(UpdateCommand cmd) {
     }
   }
 
-  private List<SolrCmdDistributor.Node> getCollectionUrls(
-      String collection, EnumSet<Replica.Type> types, boolean onlyLeaders) {
```

Comment:
`onlyLeaders` was always true therefore `types` was not used.

Reply:
I looked at the git history.
This code comes from a refactor in 2019 of DistributedUpdateProcessor to extract the code specific to cloud. At that time this method signature already contained the boolean.
Looking earlier, I saw that the boolean was added in 2018 in a commit “Make massive improvements to the tests”. It added the boolean and modified the single call to pass true. Well, I believe it was a perf optimization to always limit to leaders. At that time the intent was clearly to ignore the dead code, and I think the dead code should have been removed from the method instead of adding this boolean.
So after looking at the context, I’m +1 to remove this dead code.